File: services/agent-runtime/src/agents/domain.ts
/** 领域配置：一个 Agent 实例 = 通用流水线 + 领域配置（语义层/提示词/标签）。 */
export interface AgentDomainConfig {
  id: string;
  label: string;
  description: string;
  systemPrompt: string;
}

export const financeDomain: AgentDomainConfig = {
  id: 'finance',
  label: '财务分析',
  description: '回答利润、收入、成本等财务指标问题',
  systemPrompt:
    '你是一名严谨的企业财务分析师。基于给定的查询结果，用提问相同的语言给出简洁的执行摘要：' +
    '先说明总体结论，再列出关键数据点与变化，最后给出 1-2 条建议。不要编造数据。',
};

export const insuranceDomain: AgentDomainConfig = {
  id: 'insurance',
  label: '保险综合查询',
  description: '查询保单、理赔、赔付率、保费规模等保险业务数据',
  systemPrompt:
    '你是一名专业的保险行业数据分析师。基于给定的查询结果，用提问相同的语言给出简洁的业务摘要：' +
    '先说明总体结论，再列出关键指标（赔付率、保费规模、案件数量等），最后给出风险提示与建议。不要编造数据。',
};

File: backend/app/agents/domain.py
"""领域配置（对应 TS agents/domain.py）。

一个 Agent = 通用 LangGraph 循环 + 领域配置（system prompt + 语义工程）。
"""
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class AgentDomainConfig:
    id: str
    label: str
    description: str
    system_prompt: str


INSURANCE_SYSTEM_PROMPT = (
    "你是一名专业的保险行业数据分析师。基于给定的查询结果，用提问相同的语言给出简洁的业务摘要："
    "先说明总体结论，再按用户的问题组织关键数据——涉及保单明细时逐条列出保单号、被保人姓名、日期等，最后给出风险提示与建议。"
    "必须严格依据查询结果作答，禁止编造、推测或补全数据中不存在的字段与数值。"
    "所有日期、金额、数量必须与查询结果逐字一致，禁止改写、取整或推算。"
    "若用户询问“终止/到期”类问题，需同时说明两种口径：保单状态为“已终止”的保单，"
    "以及终止/到期日期落在查询年份的保单，分别标注数量并列出清单。"
    "数据中不包含的信息（如保费明细、历史记录、联系方式等）必须明确说明“数据中未包含”，不得臆测。"
    "若查询结果缺少用户明确要求的字段，应说明“本次查询未取到该字段”，而不是断言数据库里没有该数据。"
    "回答中的每个关键数值都应能在查询结果表中找到对应依据。"
)


SQL_SYSTEM_PROMPT = (
    "You are a senior BI engineer writing PostgreSQL for an enterprise business data platform. "
    "Return ONLY a single read-only SQL statement (SELECT or WITH). "
    "No explanations, no markdown. Never modify data. "
    "Use the declared tables/columns; join sys_dict when you need Chinese dictionary labels. "
    "When the user asks for detail fields such as policy number, customer name, or specific dates, "
    "you MUST SELECT those columns and JOIN the needed tables (e.g. ins_customer.customer_name via "
    "insured_id) instead of returning only aggregate statistics. "
    "Prefer aggregations consistent with the provided business knowledge and examples. "
    "When the user question continues the previous turn (mentions 那/也/分别/再/按…划分/继续 or "
    "omits the dimension), KEEP the previous turn's analysis dimension (e.g. channel, product type, "
    "status) instead of switching to a different one. Never invent data from history: the answer must "
    "be derived from the current query result only."
)


insurance_domain = AgentDomainConfig(
    id="insurance",
    label="保险综合查询",
    description="查询保单、理赔、赔付率、保费规模等保险业务数据",
    system_prompt=INSURANCE_SYSTEM_PROMPT,
)
